Sources/MapboxMaps/Style/StyleSourceManager.swift

```diff
@@ -182,3 +220,9 @@ internal final class StyleSourceManager: StyleSourceManagerProtocol {
         backgroundQueue.async(execute: item)
     }
 }
+
+private extension SettingsServiceInterface {
+    var shouldUseDirectGeoJSONUpdate: Bool {
```

Comment:
should we log the error if the associated value is missing or is of wrong type?
```
var shouldUseDirectGeoJSONUpdate: Bool {
  do {
    return try self.get(key: "geojson_allow_direct_setter", type: Bool.self).get()
  } catch {
    Log.error(forMessage: "Failed to get value for key geojson_allow_direct_setter: \(error)")
    return false
  }
}
```

Reply:
Looking good 👍 I'm leaving the logging out though, as absence of this setting is not an error.